floodmodeller_api/toolbox/model_build/structure_log/structure_log.py

```diff
@@ -19,237 +53,445 @@ def _add_fields(self):
+        }
+        return {"dimensions": dimensions, "friction": friction}
 
-    def _rectangular_data(self, conduit, length):
-        dimensions = f"h: {conduit.height:.2f} x w: {conduit.width:.2f} x l: {length:.2f}"
-        all_mannings = [
+    def _rectangular_data(self, conduit):
+        dimensions = extract_attrs(conduit, {"width", "height", "invert"})
+        all_friction = [
             conduit.friction_on_invert,
             conduit.friction_on_soffit,
             conduit.friction_on_walls,
         ]
-        mannings_set = {min(all_mannings), max(all_mannings)}
-        if len(mannings_set) == 1:
-            friction = f"Mannings: {mannings_set.pop()}"
-        else:
-            friction = f"Mannings: [min: {mannings_set.pop()}, max: {mannings_set.pop()}]"
+        friction_set = sorted(set(all_friction))
+        friction = {
+            "friction_eq": conduit.friction_eq,
+            "friction_set": friction_set,
+            "all_friction": all_friction,
+        }
 
-        return [friction, dimensions]
+        return {"dimensions": dimensions, "friction": friction}
+
+    def _section_data(self, conduit):
+        # Symmetrical conduits
+        # these have a lot of weirdness in terms of data validity and FM will rearrange data that it thinks is wrong; so its mostly worth just trusting the modeller here
 
-    def _section_data(self, conduit, length):
         x_list = conduit.coords.x.tolist()
-        width = (max(x_list) - min(x_list)) * 2
         y_list = conduit.coords.y.tolist()
+        all_friction = conduit.coords.cw_friction.tolist()
+        # want to serialise coords so converting out of dataframe
+        width = max(x_list) * 2
         height = max(y_list) - min(y_list)
-        # currently this means that height goes to the top of the spike,
-        # it is only meant to go up to the height of the majority of the area
-        dimensions = f"h: {height:.2f} x w: {width:.2f} x l: {length:.2f}"
-        all_cw_frictions = conduit.coords.cw_friction.tolist()
-        cw_frictions_set = {min(all_cw_frictions), max(all_cw_frictions)}
-        if len(cw_frictions_set) == 1:
-            friction = f"Colebrook-White: {cw_frictions_set.pop()}"
-        else:
-            friction = (
-                f"Colebrook-White: [min: {cw_frictions_set.pop()}, max: {cw_frictions_set.pop()}]"
-            )
+        elevation_invert = min(y_list)
+        dimensions = {
+            "width": width,
+            "height": height,
+            "invert": elevation_invert,
+            "section_data": {"x": x_list, "y": y_list},
+        }
 
-        return [friction, dimensions]
+        all_friction = conduit.coords.cw_friction.tolist()
+        friction_set = sorted(set(all_friction))
+        friction = {
+            "friction_eq": "COLEBROOK-WHITE",
+            "friction_set": friction_set,
+            "all_friction": all_friction,
+        }  # friction should be CW here; should test this
```

Comment:
did you test it?

Reply:
Yes, FM only allows for colebrook-white friction for section conduits; although I did possibly identify an issue with the default conduit parameters in the API, related to this